test-data/unit/check-dataclasses.test

```diff
@@ -1420,4 +1425,130 @@ class Some:
 class Other:
     __slots__ = ('x',)
     x: int
+
+@dataclass(slots=False)  # E: Keyword argument "slots" for "dataclass" is only valid in Python 3.10 and higher
+class Third:
+    x: int
+[builtins fixtures/dataclasses.pyi]
+
+
+[case testDataclassWithUnsafeHashFalse]
+# flags: --python-version 3.7
+from dataclasses import dataclass
+
+@dataclass(unsafe_hash=False, eq=False, frozen=False)
+class FirstCase1:
+    x: int
+reveal_type(FirstCase1(1).__hash__)  # N: Revealed type is "def () -> builtins.int"
+
+@dataclass(unsafe_hash=False, eq=False, frozen=False)
+class FirstCase2:
+    x: int
+reveal_type(FirstCase2(1).__hash__)  # N: Revealed type is "def () -> builtins.int"
```

Comment:
This looks identical to `FirstCase1`. Is this redundant or should this be different?

Reply:
Indeed it was. Fixed! I've also found a bunch of missed cases.